File: language-samples/hello.ml
(* OCaml sample for GitHub language detection
   BarbrickDesign - Complete Language Portfolio *)

type greeter = {
  message : string;
}

let greet g =
  print_endline g.message

let main () =
  let greeter = { message = "Hello from OCaml!" } in
  greet greeter;
  
  (* List *)
  let numbers = [1; 2; 3; 4; 5] in
  let doubled = List.map (fun x -> x * 2) numbers in
  
  (* Record *)
  type language_info = {
    language : string;
    paradigm : string;
    typing : string;
  }
  
  let info = {
    language = "OCaml";
    paradigm = "Functional";
    typing = "Static";
  } in
  
  (* Higher-order functions *)
  let languages = ["ocaml"; "f#"; "haskell"] in
  let upper = List.map String.uppercase_ascii languages in
  
  (* Pattern matching *)
  let describe x =
    match x with
    | 42 -> "The answer"
    | _ -> "Something else"
  in
  
  List.iter (Printf.printf "%d ") doubled;
  print_newline ();
  List.iter (Printf.printf "%s ") upper;
  print_newline ()

let () = main ()

File: language-samples/hello.fs
# F# sample for GitHub language detection
# BarbrickDesign - Complete Language Portfolio

type Greeter(message: string) =
    member this.Message = message
    member this.Greet() = printfn "%s" this.Message

let main () =
    let greeter = Greeter("Hello from F#!")
    greeter.Greet()
    
    // List
    let numbers = [1; 2; 3; 4; 5]
    let doubled = numbers |> List.map (fun x -> x * 2)
    
    // Record
    type LanguageInfo = {
        Language: string
        Paradigm: string
        Typing: string
    }
    
    let info = {
        Language = "F#"
        Paradigm = "Functional"
        Typing = "Static"
    }
    
    // Pipeline
    let languages = ["f#"; "ocaml"; "haskell"]
    languages
    |> List.map (fun s -> s.ToUpper())
    |> List.iter (printfn "%s")
    
    // Pattern matching
    let describe x =
        match x with
        | 42 -> "The answer"
        | _ -> "Something else"
    
    printfn "%A" doubled
    printfn "%s" (describe 42)

main()
